src/MauiIcons.Core/Controls/MauiIcon.Attached.cs

```diff
@@ -0,0 +1,158 @@
+﻿using MauiIcons.Core.Base;
+using MauiIcons.Core.Converters;
+using MauiIcons.Core.Helpers;
+
+namespace MauiIcons.Core;
+public partial class MauiIcon
+{
+    public static new readonly BindableProperty IconProperty = BindableProperty.CreateAttached("Icon", typeof(BaseIcon), typeof(MauiIcon), null, propertyChanged: OnIconPropertyChanged);
+
+    private static void OnIconPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (newValue is null || newValue is not BaseIcon)
+            return;
+
+        BaseIcon baseIcon = (BaseIcon)newValue;
```

Comment:
@AathifMahir here you could set `baseIcon.SetBinding(BindableObject.BindingContextProperty, new Binding(nameof(BindingContext), source: bindable)` to propagate the binding context.